File: 21. Merge Two Sorted Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        if(nullptr==l1) return l2;
        if(nullptr==l2) return l1;
        if(l1->val>l2->val) swap(l1,l2);
        ListNode* head=l1;
        while(l1->next&&l2){
            if(l1->next->val>l2->val){
                ListNode* tmp=l1->next;
                l1->next=l2;
                l2=l2->next;
                l1->next->next=tmp;
                l1=l1->next;
            }else{
                l1=l1->next;
            }
        }
        if(l2){
            l1->next=l2;
        }
        return head;
    }
};

File: 21. Merge Two Sorted Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        if (l1 == null) return l2;
        if (l2 == null) return l1;
        
        ListNode dummyHead = new ListNode(0);
        ListNode current = dummyHead;
        
        while (l1 != null && l2 != null){
            if (l1.val <= l2.val){
                current.next = l1;
                current = current.next;
                l1 = l1.next;
            }
            else{
                current.next = l2;
                current = current.next;
                l2 = l2.next;
            }
        }
        while (l1 != null){
            current.next = l1;
            current = current.next;
            l1 = l1.next;
        }
        while (l2 != null){
            current.next = l2;
            current = current.next;
            l2 = l2.next;
        }
        
        return dummyHead.next;
    }
}